R/new_covariate.R

```diff
@@ -70,21 +71,19 @@ new_covariate <- function(
       if(! (0 %in% new_times)) {
         # add zero time, and remove all t < 0
         if(any(new_times > 0)) { # add the interpolated obs before and after 0 as t=0
-          if(class(new_values) %in% c("numeric", "integer")) {
-            if(implementation == 'interpolate') {
-              new_times <- c(new_times, 0)
-              y1 <- utils::tail(new_values[new_times < 0],1)
-              y2 <- utils::head(new_values[new_times > 0],1)
-              t1 <- utils::tail(new_times[new_times < 0],1)
-              t2 <- utils::head(new_times[new_times > 0],1)
-              grad <-  (y2-y1) / (t2-t1)
-              new_values <- c(new_values, y1 + grad * (0-t1))
-              new_unit <- c(new_unit, utils::tail(new_unit[new_times < 0],1))
-            } else { # add the last obs before 0 as t=0
-              new_times <- c(new_times, 0)
-              new_values <- c(new_values, utils::tail(new_values[new_times < 0], 1))
-              new_unit <- c(new_unit, utils::tail(new_unit[new_times < 0], 1))
-            }
+          if(mode(new_values) == "numeric" && implementation == "interpolate") {
```

Comment:
Earlier in the code at line 43 we have this logical condition:
```
if(implementation == "interpolate" && inherits(values, "numeric") && !is.null(interpolation_join_limit) && interpolation_join_limit > 0)
```

`mode(new_values) == "numeric"` will return `TRUE` for both doubles and integers; `inherits(values, "numeric")` will return `FALSE` for integers. Should we use `mode(values) == "numeric"` up at line 43 too for consistency?

Reply:
I've updated accordingly. Since we now also switch to LOCF if the values are non-numeric, this check was also greatly simplified.